README.md

```diff
@@ -108,6 +108,16 @@ then additional inputs/outputs should be submitted to the x-common repository.
 Changes to the test suite (style, boilerplate, etc) will probably have to be made to
 `example.tt`.
 
+### Exercise Generators
+
+If you wish to create a new generator, or edit an existing one, the generators currently live in the lib directory and are named `$PROBLEM_cases.rb`.  For example, the hamming generator is `lib/hamming_cases.rb`.  
+
+All generators currently adhere to a common public interface, and must define the following three methods:
+
+- `test_name` - Output the name of the test
+- `workload` - Output the body of the test
```

Comment:
Any thoughts about what it should do?  Just return an assertion?  If we are just returning an assertion than we might need to consider defining additional methods (i.e. `input` and `expected`.  I think it boils down to what belongs in the template vs the cases class.

Some recent generators (tournament for example) are defining an input and expect variable in the view, but the workload method is relying on the input variable being defined in the view, which does not seem ideal to me.  

Keeping all the presentation confined to the workload method allows you to keep everything together at the same level of abstraction.  When I originally thought about the interface I was thinking the Cases classes would be almost like a presenter and workload would just spit out the test body.  Since every exercise needs to define a test body, it is really just a matter of implementing its presentation.

Anyway these are just some of my thoughts, let me know what you think.